third_party/move/move-compiler-v2/tests/reference-safety/function_ref.move

```diff
@@ -0,0 +1,37 @@
+module 0x42::m {
+    public fun g(x: &mut u64, _y: &address): &mut u64 {
```

Comment:
It would be nice if the positive and negative cases would be separated out into two files, which makes it easier to show case that we are not reporting a bug in one case for the right reasons (as opposed to being shadowed by a different error/earlier error).

Reply:
done